src/audio/comp_module/module.c

```diff
@@ -0,0 +1,329 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2019 NXP. All rights reserved.
+//
+// Author: Paul Olaru <paul.olaru@nxp.com>
+
+#include <config.h>
+
+#if CONFIG_COMP_MODULE
```

Comment:
Better just don't compile file at all if you don't need this feature.
f.e. something like that in cmake file->
```
if(CONFIG_COMP_MODULE)
	add_local_sources(sof module.c)
endif()
```

Reply:
Thanks, I actually had a form of double-guarding (CMakeFiles AND in-source). Addressed.